### PR TITLE
Implement Infinite Scroll

### PR DESCRIPTION
**Pull request by neverwannafly**
May-18-2020 22-30-01

### CODE REVIEW

File: instagram/app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -34,8 +34,17 @@ def edit
   def paginate
     page_num = params[:page_num].to_i - 1
     items_per_page = params[:items_per_page].to_i
-    @post = Post.order(created_at: :desc).offset(page_num*items_per_page).limit(items_per_page)
-    render json: @post
+    @posts = Post.order(created_at: :desc).offset(page_num*items_per_page).limit(items_per_page)
+    json_response = []
+    @posts.each do |post| 
+      json_response.insert(-1, {
```
Comment (ayushg007) [inline instagram/app/controllers/posts_controller.rb:40]: I think u r inserting at the end.So can we use "push" function to insert at the end ??

File: instagram/app/javascript/packs/application.js [hunk 2]
```diff
@@ -16,4 +16,61 @@ require("channels")
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
 
-import "bootstrap"
\ No newline at end of file
+import "bootstrap"
+import $ from "jquery"
```
Comment (ayushg007) [inline instagram/app/javascript/packs/application.js:20]: try to find the difference b/w `//=require` and `import ` while including libraries

File: instagram/app/javascript/packs/application.js [hunk 3]
```diff
@@ -16,4 +16,61 @@ require("channels")
...
+
+let currentPage;
+let itemsPerPage;
+let url;
+let processed;
```
Comment (ayushg007) [inline instagram/app/javascript/packs/application.js:25]: As u r using all this variables in application.js and this file is served all over ur app . Don't u think these will cause conflicts or break code if u/other define/uses the variables in other js file??

File: instagram/app/controllers/posts_controller.rb (hunk 1 above)
Comment (neverwannafly) [reply instagram/app/controllers/posts_controller.rb:40]: Yes I didnt read API of ruby particularly well, I'll keep that in mind

**Pull request merged**